### PR TITLE
Initial example for the Fetch API

### PR DESCRIPTION
**Pull request by eminence**
CC #449

### CODE REVIEW

**Comment by alexcrichton** [comment]
This looks awesome, thanks!

I think there's a permissions error on `build.sh`, but otherwise looks good to go!

Also, perhaps this could show an example of using the serde feature to deserialize the JSON as well?

File: examples/fetch/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+fetch.d.ts
+fetch.js
+fetch_bg.wasm
+package-lock.json
```
Comment (jrakow) [inline examples/fetch/.gitignore]: There is a newline missing here.

Comment (eminence) [reply examples/fetch/.gitignore]: Should be fixed now

**Comment by eminence** [comment]
Example has been updated to show off serde magic.

I would have liked for `run()` to return a Promise to a `Branch` object.  `future_to_promise` requires that the future resolve to a `JsValue` type, so while I could have returned the `Branch` as a `JsValue`, I wasn't sure how to convert that back into a `Branch` object in javascript land.  Any hints?

File: examples/fetch/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
+#![feature(use_extern_macros)]
```
[the remark this answers is not in the document]
Comment (eminence) [reply examples/fetch/src/lib.rs]: Removed \o/

**Pull request merged**